fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx

```diff
@@ -15,33 +15,56 @@ interface Props {
   onSelect: (jobCategory: AnyJobCategory, type: 'parent' | 'child') => void;
   tone: ComponentProps<typeof Dropdown>['tone'];
   hideLabel?: boolean;
+  initialValue?: string;
 }
 
 const JobCategorySelectInput = ({
   jobCategories,
   onSelect,
   hideLabel,
+  initialValue,
   ...restProps
 }: Props) => {
   const [selectedParentCategoryId, setSelectedParentCategoryId] = useState('');
   const [selectedChildCategoryId, setSelectedChildCategoryId] = useState('');
   const [childCategories, setChildCategories] = useState<AnyJobCategory[]>();
 
+  const handleParentCategorySelect = (parentCategoryId: string) => {
+    const parentCategory = findObjectByOid(jobCategories, parentCategoryId);
+
+    setSelectedParentCategoryId(parentCategoryId);
+
+    if (parentCategory?.children) {
+      setChildCategories(parentCategory.children);
+      setSelectedChildCategoryId('');
+      onSelect(parentCategory, 'parent');
+    }
+  };
+
   useEffect(() => {
-    if (selectedParentCategoryId !== '') {
-      const parentCategory = findObjectByOid(
-        jobCategories,
-        selectedParentCategoryId,
-      );
+    if (initialValue) {
+      for (const parentCategory of jobCategories) {
+        if (!parentCategory.children) {
+          continue;
+        }
 
-      if (parentCategory?.children) {
-        setChildCategories(parentCategory.children);
-        setSelectedChildCategoryId('');
-        onSelect(parentCategory, 'parent');
+        if (parentCategory.id.value === initialValue) {
+          setChildCategories(parentCategory.children);
+          setSelectedParentCategoryId(parentCategory.id.value);
+          return;
+        }
+
+        for (const childCategory of parentCategory.children) {
+          if (childCategory.id.value === initialValue) {
+            setChildCategories(parentCategory.children);
+            setSelectedParentCategoryId(parentCategory.id.value);
+            setSelectedChildCategoryId(childCategory.id.value);
+            return;
+          }
+        }
       }
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [selectedParentCategoryId]);
+  }, [initialValue, jobCategories]);
 
   useEffect(() => {
```

Comment:
Can this be a handler as well for consistency? AFAICS it has the same behaviour as the parent handler. It'd also be good to group them in the source file instead of having the `initialValue` handling splitting them up.

Reply:
Also good point. I ignored that useEffect 🤦